sqlglot/parser.py

```diff
@@ -1962,18 +1981,17 @@ def _match_set(self, types):
 
         return None
 
-    def _match_pair(self, token_type_a, token_type_b):
-        if not self._curr or not self._next:
-            return None
-
-        if (
-            self._curr.token_type == token_type_a
-            and self._next.token_type == token_type_b
-        ):
-            self._advance(2)
-            return True
-
-        return None
+    def _match_sequence(self, *sequence, advance=True):
+        for i, sequence_token_type in enumerate(sequence):
+            actual_token = self._get_token_at_index(self._index + i)
+            if not actual_token or sequence_token_type != actual_token.token_type:
+                return False
+        if advance:
+            self._advance(len(sequence))
```

Comment:
you may actually be able to just do self._advance(i)

Reply:
`self._advance(i+1)` technically works but IDE isn't happy with it. I think the problem is that in the case where sequence is empty then the i never gets initialized. That really shouldn't happen in practice but going to keep the current way since it seems comparable and removes a warning.

Reply:
what if you set i = 0?

Reply:
Yeah but now we are adding another line of code... seems all messier then this approach.

Reply:
as long as there's no performance implication

Reply:
i'm actually a bit afraid that _match_sequence is gonna significantly slow down the parser. you're now dynamically getting grabbing tokens instead of using the cache that was previously used 

self._curr and self._next

was this change necessary? do you even match more than just the pair?

Reply:
This goes back to the typo that you mentioned. Originally I was matching on 3 things but then realized later I only needed 2. So yeah I can remove this now since it is not currently needed. Once someone wants to match on >2 items then we will need it.

Reply:
cool yea let’s revert it for now. in general if you need to match things like that you can also use the tokenizer like with time zone